apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -148,45 +144,68 @@ export class UpsertWorkflowUseCase {
       __source: workflowDto.__source || WorkflowCreationSourceEnum.DASHBOARD,
       type: WorkflowTypeEnum.BRIDGE,
       origin: WorkflowOriginEnum.NOVU_CLOUD,
-      steps: this.mapSteps(workflowDto.steps),
+      steps,
       active: isWorkflowActive,
       description: workflowDto.description || '',
       tags: workflowDto.tags || [],
       userPreferences: command.workflowDto.preferences?.user ?? null,
       defaultPreferences: command.workflowDto.preferences?.workflow ?? DEFAULT_WORKFLOW_PREFERENCES,
       triggerIdentifier: slugify(workflowDto.name),
+      status: this.computeStatus(command.workflowDto, steps),
     };
   }
 
-  private convertCreateToUpdateCommand(
+  private computeStatus(workflow: UpsertWorkflowDataCommand, steps: NotificationStep[]) {
+    const hasIssues = steps.filter((step) => this.hasControlIssues(step.issues)).length > 0;
```

Comment:
Nit picky one, please compute `hasIssues`, after checking on the active flag to avoid unnecessary computations for inactive workflows.

Reply:
done! :)